File: Dante1/Module1.07/Zadanie7.15/main.c
#include <stdio.h>
#include <math.h>

float toRadian(float a);
float toDegree(float a);

int main()
{
    int type;
    printf("Enter type: ");
    int check = scanf("%i", &type);
    if(check != 1)
    {
        printf("Incorrect input");
        return 1;
    }
    else if(type == 1)
    {
        float kat;

        printf("Enter kat: ");
        int check1 = scanf("%f",&kat);
        if(check1 != 1)
        {
            printf("Incorrect input");
            return 1;
        }

        printf("Result: %.2f", toRadian(kat));
    }
    else if(type == 2)
    {
        float kat;

        printf("Enter kat: ");
        int check1 = scanf("%f",&kat);
        if(check1 != 1)
        {
            printf("Incorrect input");
            return 1;
        }

        printf("Result: %.2f", toDegree(kat));
    }
    else
    {
        printf("Incorrect input ");
        return 1;
    }
    return 0;
}

float toRadian(float a)
{
    return a * M_PI / 180;
}

float toDegree(float a)
{
    return a * 180 / M_PI;
}

File: Dante1/Module1.07/Zadanie7.15/main.rs
fn toRadian(a: f32) -> f32 {
    a * std::f32::consts::PI / 180.0
}

fn toDegree(a: f32) -> f32 {
    a * 180.0 / std::f32::consts::PI
}

fn main() {
    let mut input = String::new();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("err");
    let what: i32 = input.trim().parse().expect("err");
    if what == 1 {
        input.clear();
        println!("Enter: ");
        std::io::stdin().read_line(&mut input).expect("err");
        let kat: f32 = input.trim().parse().expect("err");
        println!("{}", toRadian(kat));
    }
    else if what == 2 {
        input.clear();
        println!("Enter: ");
        std::io::stdin().read_line(&mut input).expect("err");
        let kat: f32 = input.trim().parse().expect("err");
        println!("{}", toDegree(kat));
    }
    else {
        panic!("err");
    }
}
